iceoryx_posh/include/iceoryx_posh/internal/popo/ports/base_port_data.hpp

```diff
@@ -44,19 +42,19 @@ struct BasePortData
     /// @param[in] portType Type of port to be created
     /// @param[in] processName Name of the process
     /// @param[in] runnable The runnable where this port is attached to
-    BasePortData(const capro::ServiceDescription& serviceDescription,
-                 const cxx::CString100& processName) noexcept;
+    BasePortData(const capro::ServiceDescription& serviceDescription, const ProcessName_t& processName) noexcept;
 
     BasePortData(const BasePortData&) = delete;
     BasePortData& operator=(const BasePortData&) = delete;
     BasePortData(BasePortData&&) = delete;
     BasePortData& operator=(BasePortData&&) = delete;
+    ~BasePortData() = default;
 
     capro::ServiceDescription m_serviceDescription;
-    cxx::CString100 m_processName;
+    ProcessName_t m_processName;
 
     static std::atomic<uint64_t> s_uniqueIdCounter;
-    const std::atomic<uint64_t> m_uniqueId{0};
+    const std::atomic<uint64_t> m_uniqueId;
```

Comment:
Why this change?

Reply:
same same

Reply:
I will give this a try `const std::atomic<uint64_t> m_uniqueId{s_uniqueIdCounter.fetch_add(1u, std::memory_order_relaxed)};`